File: gaia-sdk-javascript/src/graphql/request/type/CreatedSkillBuildJobImpulse.ts

import {SkillBuildJob} from "./SkillBuildJob";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

export class CreatedSkillBuildJobImpulse extends Array<(_:VariableRegistry) => string> {
public _typeName = "CreatedSkillBuildJobImpulse";
    public id = () => { 
        this.push(_ => "id")
    };

    public data = (config: (_:SkillBuildJob) => void) => this.push((registry) => {
        const entity = new SkillBuildJob();
        config(entity);
        return "data { " + entity.render(registry) + " }";
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/request/type/CreatedSkillBuildJobImpulse.py

from gaia_sdk.graphql.request.type.SkillBuildJob import SkillBuildJob

from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class CreatedSkillBuildJobImpulse(list):

    def id(self):
        self.append(lambda x: "id")

    def data(self, config: Callable[['SkillBuildJob'], None]):
        def callback(registry: VariableRegistry):
            entity = SkillBuildJob()
            config(entity)
            return "data {" + entity.render(registry) + "}"
        self.append(callback)

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))
